Fix attrs 25.2.0 compatibility

### Body

attrs just released 25.2.0, which was broken Airflow with exceptions like:

```
    self.dag = DAG("test_dag_id", default_args=args)
E   TypeError: __init__() got an unexpected keyword argument 'default_args'
```

https://github.com/python-attrs/attrs/releases/tag/25.2.0

Might be this change: https://github.com/python-attrs/attrs/releases/tag/25.2.0

Upstream issue: https://github.com/python-attrs/attrs/issues/1416

### Committer

- [x] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/hatch_build.py
+++ b/hatch_build.py
@@ -191,9 +191,7 @@ DEPENDENCIES = [
     "alembic>=1.13.1, <2.0",
     "argcomplete>=1.10",
     "asgiref>=2.3.0",
-    # attrs 25.2.0 has broken DAG __init__
-    # tracking issue: https://github.com/apache/airflow/issues/47682
-    "attrs>=22.1.0, <25.2.0",
+    "attrs>=22.1.0, !=25.2.0",
     # Blinker use for signals in Flask, this is an optional dependency in Flask 2.2 and lower.
     # In Flask 2.3 it becomes a mandatory dependency, and flask signals are always available.
     "blinker>=1.6.2",

--- a/task-sdk/pyproject.toml
+++ b/task-sdk/pyproject.toml
@@ -22,9 +22,7 @@ description = "Python Task SDK for Apache Airflow DAG Authors"
 readme = { file = "README.md", content-type = "text/markdown" }
 requires-python = ">=3.9, <3.13"
 dependencies = [
-    # attrs 25.2.0 has broken DAG __init__
-    # tracking issue: https://github.com/apache/airflow/issues/47682
-    "attrs>=24.2.0, <25.2.0",
+    "attrs>=24.2.0, !=25.2.0",
     "httpx>=0.27.0",
     "jinja2>=3.1.4",
     "methodtools>=0.4.7",

